src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let clone = Object.assign({}, state);
+  for (let i = 0; i < actions.length; i++) {
+    const { type, extraData, keysToRemove } = actions[i];
+
+    if (i > 0) {
+      clone = Object.assign({}, newState[i - 1]);
+    }
+
+
+    if (type === 'addProperties') {
+      newState.push({ ...clone, ...extraData });
```

Comment:
Instead of pushing the new state inside the condition, you should perform this action once after the switch statement to adhere to the DRY principle. ([DONT REPEAT YOURSELF] - If you perform same action in all `switch` cases - do it just once afterwards.)